### ISSUE TITLE
No rule for snake_case with number

### ISSUE THREAD

**Issue by jessiel-hacke** [bbatsov/ruby-style-guide#606]
There is no rule for a variable name with numbers.

For example if I have a variable named like `a_400`, should it be `a400` instead?

Having

```
# bad
a_400 = something

# good
a400 = something
```

or both situations are good?

**Comment by RomanADavis** [comment]
Wouldn't it be better just to avoid numbers in variable names? If you need a1 and an a2, wouldn't you rather make an array a with a[1] and an a[2]?

Am I missing a use case here?

**Comment by RomanADavis** [comment]
Wouldn't it be better just to avoid numbers in variable names? If you need a1 and an a2, wouldn't you rather make an array a with a[1] and an a[2]?

Am I missing a use case here?

**Comment by RomanADavis** [comment]
Wouldn't it be better just to avoid numbers in variable names? If you need a1 and an a2, wouldn't you rather make an array a with a[1] and an a[2]?

Am I missing a use case here?

**Comment by RomanADavis** [comment]
Wouldn't it be better just to avoid numbers in variable names? If you need a1 and an a2, wouldn't you rather make an array a with a[1] and an a[2]?

Am I missing a use case here?

**Comment by RomanADavis** [comment]
Wouldn't it be better just to avoid numbers in variable names? If you need a1 and an a2, wouldn't you rather make an array a with a[1] and an a[2]?

Am I missing a use case here?

**Comment by RomanADavis** [comment]
Wouldn't it be better just to avoid numbers in variable names? If you need a1 and an a2, wouldn't you rather make an array a with a[1] and an a[2]?

Am I missing a use case here?

**Comment by RomanADavis** [comment]
Wouldn't it be better just to avoid numbers in variable names? If you need a1 and an a2, wouldn't you rather make an array a with a[1] and an a[2]?

Am I missing a use case here?

**Comment by jessiel-hacke** [comment]
I think there are a few use cases where I could use this format of variable naming. As an example, cnab400, which is a format for bank files used on batch payments. Maybe I should use a different name for the variable but it would be nice to have it on the style guide saying that I should use a different format.

**Comment by eliotsykes** [comment]
**TLDR** In the open source Ruby projects searched, not using underscores in numbered variables happens more frequently (e.g. `a1` is used more often than `a_1`).

---

Without opinion, in case it helps, here's some (probably flawed) stats about usage of `_1` vs `1` suffixes, taken from [searching 100+ ruby codebases](/eliotsykes/real-world-rails):

```
# Underscored numbered variable name suffixes frequency
$ ag --ruby --stats-only '\W[a-z]_1 ='
1 matches
1 files contained matches
75948 files searched

# Not underscored numbered variable name suffixes frequency
$ ag --ruby --stats-only '\W[a-z]1 ='
1052 matches
408 files contained matches
75948 files searched
```

**Comment by jessiel-hacke** [comment]
I will open a PR to add it

**Comment by strika** [comment]
@eliotsykes, thanks for those stats. I find this very interesting. I actually use the style where numbers are separated with `_`. It's because I find this consistent with snake case variable naming - `project_one` and `project_1`. That's why I feel that not using `_` is against the snake case variable naming.

Wikipedia's article about snake case doesn't mention number specifically:

> Snake case (or snake_case) is the practice of writing compound words or phrases in which the elements are separated with one underscore character (_) and no spaces, with each element's initial letter usually lowercased within the compound and the first letter either upper or lower case—as in "foo_bar" and "Hello_world".

**Comment by eliotsykes** [comment]
@strika I suspect those stats are flawed and understate how many projects use the `*_1` style. You might want to try searching those codebases yourself and experimenting with the search regex I used above which isn't comprehensive.

**Comment by strika** [comment]
I cloned part of the project, since the space on the VM disk was filled, and I searched through 56789 Ruby files. My stats are as follows:

- 1116 - Underscored numbered variable name suffixes frequency
- 10731 - Not underscored numbered variable name suffixes frequency

So even if your stats were flawed, the non underscored style is still much more popular.

**Comment by jessiel-hacke** [comment]
https://github.com/bbatsov/ruby-style-guide/pull/609

**Comment by bbatsov** [comment]
The related PR was merged.
